lib/tools/apk-utils.js

```diff
@@ -241,21 +241,25 @@ apkUtilsMethods.waitForNotActivity = async function (pkg, act, waitMs = 20000) {
  */
 apkUtilsMethods.uninstallApk = async function (pkg) {
   log.debug(`Uninstalling ${pkg}`);
+  if (!this.isAppInstalled(pkg)) {
+    log.info(`${pkg} was not uninstalled, because it was not present on the device`);
+    return false;
+  }
+  let stdout;
   try {
     await this.forceStop(pkg);
-    let stdout = await this.adbExec(['uninstall', pkg], {timeout: 20000});
-    stdout = stdout.trim();
-    // stdout may contain warnings meaning success is not on the first line.
-    if (stdout.indexOf("Success") !== -1) {
-      log.info("App was uninstalled");
-      return true;
-    } else {
-      log.info("App was not uninstalled, maybe it wasn't on device?");
-      return false;
-    }
+    stdout = await this.adbExec(['uninstall', pkg], {timeout: 20000});
   } catch (e) {
     log.errorAndThrow(`Unable to uninstall APK. Original error: ${e.message}`);
   }
+  stdout = stdout.trim();
+  // stdout may contain warnings meaning success is not on the first line.
```

Comment:
Should these warnings be outputted?

Reply:
I do output it later if there is no Success substring there.

Reply:
But what if there are "warnings", whatever they are, but also success?

Reply:
It's not a problem at all. Added one more log.debug call